Test Str.StrToInts fails in ppc64le

Building ddnet 18.2 in Fedora Rawhide (next version 41) under architecture ppc64le, Str.StrToInts fails with the output below (-v added). 

The build runs for the 18.2 version update for all architectures can be found in: https://koji.fedoraproject.org/koji/taskinfo?taskID=117534623

A separate build with `run_tests` verbose is available in https://koji.fedoraproject.org/koji/taskinfo?taskID=117952804. Here is the full [build.log](https://github.com/ddnet/ddnet/files/15380255/build.log)

RPM spec file: [ddnet.spec](https://github.com/ddnet/ddnet/files/15380274/ddnet.spec.txt)
Pull request for updating version: https://src.fedoraproject.org/rpms/ddnet/pull-request/11

Relevant output:

> [  PASSED  ] 271 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] Str.StrToInts
 1 FAILED TEST
2024-05-20 16:38:38 I sql: SQLite statement: INSERT OR IGNORE INTO record_race(	Map, Name, Timestamp, Time, Server, 	cp1, cp2, cp3, cp4, cp5, cp6, cp7, cp8, cp9, cp10, cp11, cp12, cp13, 	cp14, cp15, cp16, cp17, cp18, cp19, cp20, cp21, cp22, cp23, cp24, cp25, 	GameId, DDNet7) VALUES ('Kobra 3', 'nameless tee', DATETIME('2021-11-24 19:24:08', 'utc'), 100.00, 'USA', 	0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 	0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 	0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 	'8d300ecf-5873-4297-bee5-95668fdff320', 0)
FAILED: CMakeFiles/run_cxx_tests /builddir/build/BUILD/ddnet-18.2/redhat-linux-build/CMakeFiles/run_cxx_tests 
cd /builddir/build/BUILD/ddnet-18.2/redhat-linux-build && /builddir/build/BUILD/ddnet-18.2/redhat-linux-build/testrunner
ninja: build stopped: subcommand failed.
RPM build errors:
error: Bad exit status from /var/tmp/rpm-tmp.LK0FV4 (%check)
    Bad exit status from /var/tmp/rpm-tmp.LK0FV4 (%check)
Child return code was: 1
EXCEPTION: [Error('Command failed: \n # /usr/bin/systemd-nspawn -q -M 5774553938f54555a29bfecf0296e517 -D /var/lib/mock/f41-build-51113557-6097625/root -a -u mockbuild --capability=cap_ipc_lock --bind=/tmp/mock-resolv.6y8ykwoq:/etc/resolv.conf --bind=/dev/btrfs-control --bind=/dev/mapper/control --bind=/dev/fuse --bind=/dev/loop-control --bind=/dev/loop0 --bind=/dev/loop1 --bind=/dev/loop2 --bind=/dev/loop3 --bind=/dev/loop4 --bind=/dev/loop5 --bind=/dev/loop6 --bind=/dev/loop7 --bind=/dev/loop8 --bind=/dev/loop9 --bind=/dev/loop10 --bind=/dev/loop11 --console=pipe --setenv=TERM=vt100 --setenv=SHELL=/bin/bash --setenv=HOME=/builddir --setenv=HOSTNAME=mock --setenv=PATH=/usr/bin:/bin:/usr/sbin:/sbin \'--setenv=PROMPT_COMMAND=printf "\\033]0;<mock-chroot>\\007"\' \'--setenv=PS1=<mock-chroot> \\s-\\v\\$ \' --setenv=LANG=C.UTF-8 --resolv-conf=off bash --login -c \'/usr/bin/rpmbuild -bb --noclean --target ppc64le --nodeps /builddir/build/SPECS/ddnet.spec\'\n', 1)]

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -291,11 +291,10 @@ if(NOT MSVC AND NOT HAIKU)
     add_cxx_compiler_flag_if_supported(OUR_FLAGS -ffloat-store)
   endif()
 
-  # This is needed to get the server to correctly display special characters
-  # on ARM systems.
-  if("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "arm" OR "${CMAKE_SYSTEM_PROCESSOR}" MATCHES "aarch64")
-    add_cxx_compiler_flag_if_supported(OUR_FLAGS -fsigned-char)
-  endif()
+  # We assume that char is signed in various places in the code. In particular,
+  # the Str.StrToInts test will fail when char is not signed and names containing
+  # special characters will be displayed incorrectly on servers.
+  add_cxx_compiler_flag_if_supported(OUR_FLAGS -fsigned-char)
 
   # Don't insert timestamps into PEs to keep the build reproducible.
   if(TARGET_OS STREQUAL "windows")

--- a/src/game/gamecore.cpp
+++ b/src/game/gamecore.cpp
@@ -9,6 +9,8 @@
 #include <base/system.h>
 #include <engine/shared/config.h>
 
+#include <limits>
+
 const char *CTuningParams::ms_apNames[] =
 	{
 #define MACRO_TUNING_PARAM(Name, ScriptName, Value, Description) #ScriptName,
@@ -62,6 +64,8 @@ float CTuningParams::GetWeaponFireDelay(int Weapon) const
 	default: dbg_assert(false, "invalid weapon"); return 0.0f; // this value should not be reached
 	}
 }
+
+static_assert(std::numeric_limits<char>::is_signed, "char must be signed for StrToInts to work correctly");
 
 void StrToInts(int *pInts, size_t NumInts, const char *pStr)
 {

